hr_attendance_calendar/models/hr_attendance_day.py

```diff
@@ -0,0 +1,297 @@
+    # due_break_total = fields.Float(string='Total break due',
+    #                                compute='_compute_due_break',
+    #                                store=True, )
+    break_ids = fields.One2many(string='Breaks',
+                                comodel_name='hr.attendance.break',
+                                inverse_name='attendance_day_id',
+                                readonly=True,
+                                store=True)
+    break_max = fields.Float(string='Longest break',
+                             compute='_compute_break_max',
+                             store=True, )
+    break_total = fields.Float(string='Total break',
+                               compute='_compute_break_total',
+                               store=True, )
+
+    # Extra hours
+    extra_hours = fields.Float(string="Extra hours",
+                               compute='_compute_extra_hours',
+                               store=True,
+                               oldname='total_bonus_malus')
+    weighting = fields.Float(String="Weighting",
+                             readonly=True)
+
+    ##########################################################################
+    #                             FIELDS METHODS                             #
+    ##########################################################################
+    @api.multi
+    @api.depends('cal_att_ids')
+    def _compute_working_day(self):
+        for att_day in self.filtered('cal_att_ids'):
+            day_id = int(self.cal_att_ids[0].dayofweek)
+            days_list = ('Monday',
+                         'Tuesday',
+                         'Wednesday',
+                         'Thursday',
+                         'Friday',
+                         'Saturday',
+                         'Sunday')
+            att_day.working_day = days_list[day_id]
+
+    @api.multi
+    @api.depends('leave_ids')
+    def _compute_in_leave(self):
+        for att_day in self:
+            if att_day.leave_ids:
+                att_day.in_leave = True
+
+    @api.multi
+    @api.depends('cal_att_ids', 'leave_ids.state')
+    def _compute_due_hours(self):
+        for att_day in self:
+            due_hours = 0
+            for cal_att in att_day.cal_att_ids:
+                due_hours += cal_att.due_hours
+
+            if att_day.leave_ids:
+                for leave in att_day.leave_ids:
+                    if leave.state not in ['validate', 'validate1']:
+                        continue
+                    if not leave.holiday_status_id.remove_from_due_hours:
+                        # todo: use config instead of 8.0
+                        due_hours -= leave.compute_work_day(
+                            att_day.date) * 8
+
+            att_day.due_hours = due_hours
+
+    @api.multi
+    @api.depends('attendance_ids.worked_hours')
+    def _compute_worked_hours(self):
+        for att_day in self.filtered('attendance_ids'):
+            total_hours = 0
```

Comment:
You can replace all with : `att_day.worked_hours = sum(att_day.attendance_ids.mapped('worked_hours'))`